tsconfig.json

```diff
@@ -17,6 +17,6 @@
     "target": "es2015",
     "module": "ESNext",
     "lib": ["esnext"],
-    "types": ["node", "jest"]
+    "types": ["node", "@jest/globals"]
```

Comment:
This one is the wrong one. This won’t work. You need to remove `@jest/globals` here and then open each test file which we are doing `jest.` and add an import statement.
```
import { jest } from ‘@jest/globals’;
```
An example is https://github.com/kulshekhar/ts-jest/blob/7a7aba90ca39c07666bfcb707098dccd7b58b668/e2e/ast-transformers/hoist-jest/__tests__/import-jest.test.js#L1

Reply:
Previously when we have typing package `@types/jest`, typescript compiler understands `jest` as namespace. Now we don’t have that namespace anymore so it throws errors and the package `@jest/globals` doesn’t expose namespace

Reply:
@ahnpnl sounds like a catastrophic change for anyone who's using Jest — augment every single test file with an explicit `import ...`, doesn't it?

See the repo I mentioned: https://github.com/noomorph/tsc-jest-28-issue

The problem I'm seeing is that `describe`, `beforeEach` are not seen as globals for some reason.

Reply:
Another way is something like 
`declare var jest: typeof import(‘@jest/globals).jest;`

Reply:
What about the other global vars?

Reply:
Do you mean I have to prepend to every test file a line like:

```ts
import { describe, test, expect, jest } from '@jest/globals';
```

Reply:
If we declare global vars in `d.ts`, I think we don’t have to import explicitly per test files

Reply:
@ahnpnl , okay but how does it help any other project using Jest and Typescript?

Reply:
Hi @SimenB , could you check out this repo: https://github.com/noomorph/tsc-jest-28-issue ?

At first I thought it is an issue with `ts-jest`, but now it looks like Jest 28 hasn't delivered enough global typings to make the migration from `@types/jest` smooth enough.

Reply:
@ahnpnl I'll try to add (tomorrow, hopefully) a new definition file to mitigate the issue in the scope of your project, but I think it makes sense also to look for a more generic solution to this.

Reply:
I think the best way to reduce the imports per test file is declaring a namespace globally, or declaring several global vars. It’s more like replicating what `@types/jest` does.

Reply:
@ahnpnl yeah, sort of. I'll make a draft for a typings file and submit it for a review.

But still I feel like it is not morally correct to fix it only in the scope of a single repo — partially because it doesn't sit well in my head how `@types/jest` is going to live from now on... 🤔  Will they have 28th version doing exactly what you suggested or not. Won't it create a conflict with `jest => node_modules/jest/package.json -> types` and `jest => node_modules/@types/jest -> ...` ?

If they won't do it, then still it can't be that every project has to redeclare these globals independently.

Reply:
or even better, `@jest/globals` should change to something like `vitest/globals` https://vitest.dev/config/#globals

See their package.json entry https://github.com/vitest-dev/vitest/blob/03a54e0651f51b739b7833817d14d2cbcf474ede/packages/vitest/package.json#L32 Jest can do something like this, which helps a lot to migrate

Reply:
This discussion is long, but if you want _globals_ typed you need to use `@types/jest`. I hope to land https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44365 at some point, but that hasn't happened yet. `@jest/globals` is not meant to be used in `types` in `tsconfig.json`, it's made to be explicitly imported

Reply:
@SimenB thanks for shedding light on the DT situation. So it doesn't go away, ok.